summarize/summarize_long_transcripts.py

```diff
@@ -0,0 +1,295 @@
+@click.option('-s','--server',type=str,default='',help='Server log file path')
+@click.option('-l','--long_convo_ids_file',type=str,required=True,help='long conversation ids text file path')
+def main(input_filepath: str, openai_api_key: str, num_cores: int, conversation_count: str, server: str, long_convo_ids_file: str):
+    '''Main Function'''
+    process(input_filepath, openai_api_key, num_cores, conversation_count, server, long_convo_ids_file)
+
+def process(input_filepath: str, openai_api_key: str, num_cores: int, conversation_count: str, server: str, long_convo_ids_file: str):
+    '''Summarization of Conversations'''
+
+    # logging config
+    if server == "":
+        server = join(str(Path(input_filepath).parent),"server_summarize.log")
+
+    logging.basicConfig(filename=server, filemode='w', level=logging.DEBUG,
+                        format='%(asctime)s - %(name)s - %(process)d - %(levelname)s -- %(message)s', datefmt='%d-%b-%y %H:%M:%S')
+
+    openai.api_key = openai_api_key
+
+    # load input data
+    with open(input_filepath,mode="r",encoding="utf8") as f:
+        data = json.load(f)
+    df = pandas.json_normalize(data=data["examples"],sep="-")
+
+    # set index as conversation ID
+    df.set_index(["context-context_id"],drop=True,inplace=True)
+
+    with open(long_convo_ids_file,mode="r",encoding="utf-8") as f:
+        long_convo_ids = f.read()
+    
+    long_convo_ids = [long_convo_id.strip() for long_convo_id in long_convo_ids.split("\n")]
+
+    long_convo_ids = [long_convo_ids[0]]
+
+    df = df[df.index.isin(long_convo_ids)]
+    assert isinstance(df,pandas.DataFrame)
+
+    print(len(df.index.unique()))
+
+    # make one conversations per row so that when splitting,
+    # all the converation utterances stays intact
+    cols = list(df.columns)
+    agg_dict = {}
+
+    for col in cols:
+        agg_dict.update({
+            col: lambda x: x.tolist()
+        })
+
+    df = df.groupby(df.index).agg(agg_dict)
+
+    dfs = numpy.array_split(df,num_cores)
+    parallelization_input = []
+    for dataframe in dfs:
+        dataframe = dataframe.explode(cols)
+        parallelization_input.append([dataframe,input_filepath])
+
+    # parallelization
+    with Pool(num_cores) as p:
+        parallelization_output = p.map(summarization,parallelization_input)
+
+    convo_processed_num = len(df.index.unique().to_list())
+    logging.info(f"Total number of conversations processed is {convo_processed_num}")
+    logging.info(f'Total Duration for the script: {time.strftime("%H:%M:%S", time.gmtime(perf_counter() - START_TIME))}')
+
+def summarization(input: list) -> None:
+    '''Summarization'''
+
+    df = input[0]
+    input_filepath = input[1]
+
+    indices = df.index.unique().to_list()
+    len_of_indices = len(indices)
+    i=0
+    
+    with open("./summarize/prompt1.txt", mode="r",encoding="utf8") as f:
+        prompt1 = f.read()
```

Comment:
The logic of what the prompt sections are and how they are used for needs explaining clearly somewhere, either in a flowerbox, an extended comment or maybe a README.md in the summarize folder?